Source/CloudySaveManager/Private/CloudySaveManager.cpp

```diff
@@ -0,0 +1,327 @@
+#include "CloudySaveManagerPrivatePCH.h"
+ 
+#include "CloudySaveManager.h"
+#include "PlatformFeatures.h"
+#include "GameFramework/SaveGame.h"
+#include "HttpRequestAdapter.h"
+#include "HttpModule.h"
+#include "IHttpResponse.h"
+#include "Base64.h"
+#include "AllowWindowsPlatformTypes.h"
+#include "ThirdParty/libcurl/include/Windows/curl/curl.h"
+#include "HideWindowsPlatformTypes.h"
+#include "string"
+
+static const int UE4_SAVEGAME_FILE_TYPE_TAG = 0x53415647;		// "sAvG"
+static const int UE4_SAVEGAME_FILE_VERSION = 1;
+static const FString BaseUrl = "http://127.0.0.1:8000";
+static const FString AuthUrl = "/api-token-auth/";
+static const FString SaveDataUrl = "/save-data/";
+static FString Token;
+
+// Automatically starts and shuts down when UE4 is started/closed
+void CloudySaveManagerImpl::StartupModule()
+{
+    UE_LOG(LogTemp, Warning, TEXT("CloudySaveManager started"));
+    AttemptAuthentication(TEXT("joel"), TEXT("1234")); // Token variable will be populated with robot's token
```

Comment:
Not fixed token. Token changes with every request, so have to authenticate each time.